Source/Diagnostics/WarpXOpenPMD.cpp

```diff
@@ -631,6 +589,77 @@ for (unsigned i = 0, n = particle_diags.size(); i < n; ++i) {
         particlesConvertUnits(ConvertDirection::SI_to_WarpX, pc, mass);
     }
 
+    // Gather the electrostatic potential (phi) on the macroparticles
+    if ( particle_diags[i].m_plot_phi ) {
+        WARPX_ALWAYS_ASSERT_WITH_MESSAGE(
```

Comment:
If we did all this on GPU, then the I/O backend code here would not need to be changed (which I kind of would prefer).

For now, can you make this if block a function call please so it does separate out nicely and does not make this sub-function longer? I would call it `add_phi_component` or so and also declare it in `ParticleIO.H` (and implement it in a new `ParticleIO.cpp`).